homeassistant/helpers/integration_platform.py

```diff
@@ -37,68 +37,113 @@ class IntegrationPlatform:
 
 
 @callback
-def _get_platform(
-    integration: Integration | Exception, component_name: str, platform_name: str
-) -> ModuleType | None:
-    """Get a platform from an integration."""
-    if isinstance(integration, Exception):
-        _LOGGER.exception(
-            "Error importing integration %s for %s",
-            component_name,
-            platform_name,
-        )
-        return None
-
-    #
-    # Loading the platform may do quite a bit of blocking I/O
-    # and CPU work. (https://github.com/python/cpython/issues/92041)
-    #
-    # We don't want to block the event loop for too
-    # long so we check if the platform exists with `platform_exists`
-    # before trying to load it. `platform_exists` will do two
-    # `stat()` system calls which is far cheaper than calling
-    # `integration.get_platform`
-    #
-    if not integration.platforms_exists((platform_name,)):
-        # If the platform cannot possibly exist, don't bother trying to load it
-        return None
-
-    try:
-        return integration.get_platform(platform_name)
-    except ImportError as err:
-        if f"{component_name}.{platform_name}" not in str(err):
-            _LOGGER.exception(
-                "Unexpected error importing %s/%s.py",
-                component_name,
-                platform_name,
-            )
-
-    return None
-
-
-@callback
-def _async_process_integration_platforms_for_component(
+def _async_integration_platform_component_loaded(
     hass: HomeAssistant,
     integration_platforms: list[IntegrationPlatform],
     event: EventType[EventComponentLoaded],
 ) -> None:
     """Process integration platforms for a component."""
-    component_name = event.data[ATTR_COMPONENT]
-    if "." in component_name:
+    if "." in (component_name := event.data[ATTR_COMPONENT]):
```

Comment:
This can become an event filter ?

Reply:
Its marked run immediately and a callback so filter would be the same speed